test/sslapitest.c

```diff
@@ -9410,6 +9410,68 @@ static int test_sni_tls13(void)
     SSL_CTX_free(cctx);
     return testresult;
 }
+
+/*
+ * Test that the lifetime hint of a TLSv1.3 ticket is no more than 1 week
+ * 0 = TLSv1.2
+ * 1 = TLSv1.3
+ */
+static int test_ticket_lifetime(int idx)
+{
+    SSL_CTX *cctx = NULL, *sctx = NULL;
+    SSL *clientssl = NULL, *serverssl = NULL;
+    int testresult = 0;
+    unsigned char data[16] = { 0 };
+    int version = TLS1_3_VERSION;
+
+#define TWO_WEEK_SEC (14 * 24 * 60 * 60)
+#define ONE_WEEK_SEC (7 * 24 * 60 * 60)
+
+    if (idx == 0) {
+        version = TLS1_2_VERSION;
+    }
+
+    if (!TEST_true(create_ssl_ctx_pair(libctx, TLS_server_method(),
+                                       TLS_client_method(), version, version,
+                                       &sctx, &cctx, cert, privkey)))
+        goto end;
+
+    /* Set the timeout to be more than 1 week */
+    if (SSL_CTX_set_timeout(sctx, TWO_WEEK_SEC)) {
+        /* compiler is complaing about return use */;
```

Comment:
complaining ... note two letter n and two letter i present in the word ... 

And what compiler complains about a (void)func(); usage - that makes no sense.

Reply:
Thx. (A little stressed now because of power outage during work hours).
This was the error:
```
Run make -s -j[4](https://github.com/openssl/openssl/runs/5668910685?check_suite_focus=true#step:4:4)
test/sslapitest.c: In function 'test_ticket_lifetime':
test/sslapitest.c:9440:11: error: ignoring return value of 'SSL_CTX_set_timeout', declared with attribute warn_unused_result [-Werror=unused-result]
 9440 |     (void)SSL_CTX_set_timeout(sctx, TWO_WEEK_SEC);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:288[7](https://github.com/openssl/openssl/runs/5668910685?check_suite_focus=true#step:4:7)1: test/sslapitest-bin-sslapitest.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:34[10](https://github.com/openssl/openssl/runs/5668910685?check_suite_focus=true#step:4:10): build_sw] Error 2
Error: Process completed with exit code 2.

Reply:
SSL_CTX_set_session_cache_mode is a macro calling SSL_CTX_ctrl that doesn't have annotations (no __owur) so why would it warn. It is that annotation on SSL_CTX_set_timeout that is causing the fun - but that warning is bogus because a (void) cast on the return should make it clear that you have decided to explicitly toss the result.

But then https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425 makes it clear this is deliberate. 
Not really how I would approach it. (void) is better than writing warning-work-around code like that.

Reply:
Agreed. I decided to validate the the return value is the default as part of the test.